data-access/mandatees-decisions.ts

```diff
@@ -0,0 +1,282 @@
+import { sparqlEscapeUri } from 'mu';
+import { Quad, Term, TermProperty } from '../types';
+import {
+  findFirstSparqlResult,
+  getBooleanSparqlResult,
+  getSparqlResults,
+} from '../util/sparql-result';
+import { TERM_TYPE, sparqlEscapeTermValue } from '../util/sparql-escape';
+import { MANDATARIS_STATUS } from '../util/constants';
+
+const STAGING_GRAPH = 'http://mu.semte.ch/graphs/besluiten-consumed';
+export const TERM_MANDATARIS_TYPE = {
+  type: TERM_TYPE.URI,
+  value: 'http://data.vlaanderen.be/ns/mandaat#Mandataris',
+} as Term;
+
+export async function getSubjectsOfType(
+  rdfType: Term,
+  triples: Array<Quad>,
+): Promise<Term[]> {
+  const uniqueSubjects = Array.from(
+    new Set(triples.map((quad: Quad) => sparqlEscapeUri(quad.subject.value))),
+  );
+  const queryForType = `
+    SELECT ?subject 
+      WHERE {
+        GRAPH ${sparqlEscapeUri(STAGING_GRAPH)} {
+          VALUES ?subject {
+            ${uniqueSubjects.join('\n')}
+          }
+          ?subject a ${sparqlEscapeTermValue(rdfType)}.
+      }
+    }
+  `;
+
+  const subjectsOfType = await querySudo(queryForType);
+  const results = getSparqlResults(subjectsOfType);
+  if (results.length === 0) {
+    return [];
+  }
+
+  return results.map((binding: TermProperty) => binding.subject);
+}
+
+export async function getValuesForSubjectPredicateInTarget(
+  quads: Array<Quad>,
+): Promise<Array<Quad>> {
+  const useAsValues = quads.map((quad: Quad) => {
+    return `(${sparqlEscapeTermValue(quad.subject)} ${sparqlEscapeTermValue(
+      quad.predicate,
+    )}) \n`;
+  });
+  const query = `
+    SELECT ?subject ?predicate ?object ?graph
+    WHERE {
+      GRAPH ?graph {  
+        VALUES (?subject ?predicate) {
+            ${useAsValues.join('')}
+        }
+        ?subject ?predicate ?object .
+        MINUS {
+          GRAPH ${sparqlEscapeUri(STAGING_GRAPH)} {
+            ?subject ?predicate ?object .
+          }
+        }
+      }
+    }
+  `;
+  const resultsInTarget = await querySudo(query);
+
+  return getSparqlResults(resultsInTarget) as Array<Quad>;
+}
+
+export async function isMandatarisInTarget(subject: Term) {
+  const mandatarisType = sparqlEscapeTermValue(TERM_MANDATARIS_TYPE);
+  const askIfMandataris = `
+    ASK {
+      ${sparqlEscapeTermValue(subject)} a ${mandatarisType} .
+      MINUS {
+        GRAPH ${sparqlEscapeUri(STAGING_GRAPH)} {
+          ${sparqlEscapeTermValue(subject)} a ${mandatarisType} .
+        }
+      }
+    }
+  `;
+  const result = await querySudo(askIfMandataris);
+
+  return getBooleanSparqlResult(result);
+}
+
+export async function findPersoonForMandatarisInGraph(
+  subject: Term,
+  graph: Term,
+): Promise<null | Term> {
+  const escaped = {
+    graph: sparqlEscapeTermValue(graph),
+    mandataris: sparqlEscapeTermValue(subject),
+  };
+
+  const queryForPersoon = `
+    PREFIX mandaat: <http://data.vlaanderen.be/ns/mandaat#>
+
+    SELECT ?object
```

Comment:
Should i change the values of the select to the values that are get? The object was used as we expected return type Quad but that changed now. We could do `select  ?persoon`

Reply:
yeah if you use ?person that might be a bit more clear.

Reply:
Done in last commit 👌🏻